Update to fs2 3.13.0-M5

mpilquist opened the pull request:
Ready for review (and unblocks #7708)

mpilquist committed:
Bump to fs2 3.13.0-M2

mpilquist committed:
Merge branch 'series/0.23' into topic/update-fs2-0.13

mpilquist committed:
Restore header

mpilquist committed:
Restore accidentally deleted ClientHelpers

mpilquist committed:
Fix warnings

mpilquist committed:
Scalafmt

mpilquist committed:
Remove unused UnixSockets materializers on JS/Native

mpilquist committed:
Drop recevieBufferSize setting on unix sockets

mpilquist committed:
Fix tests

mpilquist committed:
Add mima exclusion

mpilquist committed:
Scalafmt

mpilquist committed:
Import order

mpilquist committed:
Bump to fs2 3.13.0-M3

danicheg committed:
Log errors when fibers exit with an error in ResponseLogger

danicheg committed:
Merge branch 'series/0.23' into response-logger-error-logging

morgen-peschke committed:
Add QueryOps.withQueryParamValue

Also updates the implementation and scaladoc for QueryOps.+*? to delegate to withQueryParamValue

Fixes https://github.com/http4s/http4s/issues/7095

mpilquist committed:
Update to fs2 3.13.0-M5 - last milestone before needing native 0.5

mpilquist committed:
Merge branch 'series/0.23' into topic/update-fs2-0.13

mpilquist committed:
Remove unnecessary match

mpilquist committed:
Merge branch 'topic/update-fs2-0.13' of https://github.com/mpilquist/http4s into topic/update-fs2-0.13

mpilquist committed:
Update scalajs to 1.20.1

http4s-steward[bot] committed:
Update netty-buffer, netty-codec-http to 4.2.6.Final in series/0.23

mpilquist committed:
Remove explicit dependency on ip4s

armanbilge commented on ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala:
```diff
@@ -116,7 +111,8 @@ final class EmberClientBuilder[F[_]: Async: Network] private (
   def withoutTLSContext: EmberClientBuilder[F] = copy(tlsContextOpt = None)
 
   /** Sets the `SocketGroup`, a group of TCP sockets to be used in connections. */
-  def withSocketGroup(sg: SocketGroup[F]): EmberClientBuilder[F] = copy(sgOpt = sg.some)
+  @deprecated("Explicit socket groups are no longer supported", "0.23.next")
+  def withSocketGroup(sg: SocketGroup[F]): EmberClientBuilder[F] = this
```
Just calling out that this is a breaking change. Not sure how much this feature was used in the wild.

danicheg committed:
Merge pull request #7711 from http4s/update/series/0.23/netty-buffer-4.2.6.Final

Update netty-buffer, netty-codec-http to 4.2.6.Final in series/0.23

samspills committed:
fail parsing invalid whitespace around field name

test trailer headers specifically

better tests, clean up comments, rm space clause

test when crlf split across chunk

recover parse errors by returning BadRequest

clean up

fix the weird chunking test

handle header parse error in runConnection as well

http4s-steward[bot] committed:
Update sbt-doctest to 0.11.4 in series/0.23

rossabaker committed:
Merge pull request #7712 from http4s/update/series/0.23/sbt-doctest-0.11.4

mpilquist committed:
Address review comments

mpilquist committed:
Retarget base branch to topic/0.23/sn-0.5

mpilquist committed:
Remove unused nowarns

mpilquist committed:
Remove unused nowarns

mpilquist committed:
Remove unused nowarns

mpilquist committed:
Scalafix

samspills committed:
delete unused colon value

samspills committed:
scalafixAll

samspills committed:
cannot optimize actually

morgen-peschke committed:
Add missing type hints

morgen-peschke committed:
Appease Scalafix

rossabaker committed:
Merge pull request #7702 from morgen-peschke/ISSUE-7095-with-query-param-value

samspills committed:
Merge pull request #7716 from http4s/sam/invalid-whitespace-fix

fail parsing invalid whitespace around field name

rossabaker committed:
Merge pull request #7678 from danicheg/response-logger-error-logging

rossabaker committed:
Release v0.23.31

samspills committed:
remove [bot] so generate-site doesn't fail

samspills committed:
details block

samspills committed:
Merge pull request #7717 from http4s/release-v0.23.31

Release v0.23.31

mpilquist committed:
Target 0.23.32 release now

mpilquist committed:
Merge branch 'series/0.23' into topic/update-fs2-0.13

mpilquist committed:
Fix newly created doc warnings

mpilquist commented:
@rossabaker This one should be good to merge in to #7708 unless you see anything else outstanding
